coding/wml-v20/src/main/java/org/n52/sos/aquarius/OmObservationMapper.java

```diff
@@ -97,4 +87,20 @@ public String toInterpolationTypeHref(final OmObservation observation) {
         return wml2InterpolationHref;
     }
 
+    private String mapProcedureIdentifierToInterpolationType(String procedureIdentifier) {
+        String aquariusInterpolationType = null;
+        if (null != procedureIdentifier && !procedureIdentifier.isEmpty()) {
+            int indexOfSeparationCharacter = procedureIdentifier.lastIndexOf(SEPARATOR_CHARACTER);
+            if (indexOfSeparationCharacter > 0) {
```

Comment:
This is a strange edge case, but 0 is a legal value for lastIndexOf. If the character is not is returns -1. Maybe it's better to make this comparator >=